### PR TITLE
Improve readability of tests involving mime messages

### PR DESCRIPTION
**Pull request by chibenwa**
Because this was a hudge pain...

Because I want to be able to do simple tests on mailets with PVN...

Take it as a gift. If interesting, we create a ticket for this and merge it.

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by aduprat** [comment]
Can you create a corresponding Apache JIRA, the PVN issue number doesn't make sense here.

File: mailet/base/src/test/java/org/apache/mailet/base/test/MimeMessageBuilder.java [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+    public static class MultipartBuilder {
+        private ImmutableList.Builder<BodyPart> bodyParts = ImmutableList.builder();
+
+        public MultipartBuilder addBody(BodyPart bodyPart) {
+            this.bodyParts.add(bodyPart);
+            return this;
+        }
+
+        public MultipartBuilder addBodys(BodyPart... bodyParts) {
```
Comment (aduprat) [inline mailet/base/src/test/java/org/apache/mailet/base/test/MimeMessageBuilder.java]: addBodies

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java [hunk 2]
```diff
@@ -77,6 +81,8 @@ public void compile(Object[][] patterns) throws PatternSyntaxException {
             //Loop through the header values
             if (headers != null) for (String header : headers) {
                 if (pattern.matcher(header).matches()) {
+                    System.out.println(headerName);
```
Comment (aduprat) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java]: to remove

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/MailAttributesToMimeHeadersTest.java [hunk 3]
```diff
@@ -84,10 +84,10 @@ public void shouldAddAttributeIntoHeadersWhenHeaderAlreadyPresent() throws Messa
                 .setProperty("simplemapping", MAIL_ATTRIBUTE_NAME1 + "; " + HEADER_NAME1)
                 .build();
         mailet.init(mailetConfig);
-        
-        MimeMessage mimeMessage = MailUtil.createMimeMessage();
-        mimeMessage.addHeader(HEADER_NAME1, "first value");
-		FakeMail mockedMail = MailUtil.createMockMail2Recipients(mimeMessage);
+
+		FakeMail mockedMail = MailUtil.createMockMail2Recipients(MimeMessageBuilder.mimeMessageBuilder()
```
Comment (aduprat) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/MailAttributesToMimeHeadersTest.java]: indent

File: mailet/base/src/test/java/org/apache/mailet/base/test/MimeMessageBuilder.java (hunk 1 above)
Comment (quynhn) [inline mailet/base/src/test/java/org/apache/mailet/base/test/MimeMessageBuilder.java]: Done

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java (hunk 2 above)
Comment (quynhn) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java]: Done

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/MailAttributesToMimeHeadersTest.java (hunk 3 above)
Comment (quynhn) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/MailAttributesToMimeHeadersTest.java]: Done

**Comment by quynhn** [comment]
```
Can you create a corresponding Apache JIRA, the PVN issue number doesn't make sense here.
```
I created: https://issues.apache.org/jira/browse/MAILET-145

**Comment by chibenwa** [comment]
Duplicate of https://github.com/linagora/james-project/pull/549

**Pull request closed without merging**